### PR TITLE
front: fix map waypoints not displayed in speedspace chart an output …

### LINKED ISSUE #8845: front: display the via added by click on the map in the GEV
### What happened?

via added by click ont the map are not displayed in the GEV

### What did you expect to happen?

We should see them in the GEV

### How can we reproduce it (as minimally and precisely as possible)?

Try to create any train schedule with some via added bly click on the map

### On which environments the bug occurs?

Recette (SNCF)

### On which browser the bug occurs?

Google Chrome

### OSRD version (top right corner `Account` button > `Informations`)

bc8b913

### LINKED ISSUE #8846: front: display the via added by click on the map in the output table
### What happened?

via added by click ont the map are not displayed in the output table

### What did you expect to happen?

We should see them in the output table

### How can we reproduce it (as minimally and precisely as possible)?

Try to create any train schedule with some via added bly click on the map

### On which environments the bug occurs?

Recette (SNCF)

### On which browser the bug occurs?

Google Chrome

### OSRD version (top right corner `Account` button > `Informations`)

bc8b913

### PR DESCRIPTION
**Pull request by SharglutDev**
…table

Add an util to centralize the insertion of map waypoints in an operational points array

fix #8845
fix #8846

### CODE REVIEW

File: front/src/applications/operationalStudies/utils.ts [hunk 1]
```diff
@@ -155,35 +158,85 @@ export const formatElectrificationRanges = (
     };
   });
 
+/**
+ * Check if the train path used waypoints added by map click and add them to the operational points
+ */
+export const upsertMapWaypointsInOperationalPoints = (
```
Comment (clarani) [inline front/src/applications/operationalStudies/utils.ts]: I think you can create a new helper file for this function, the utils file is already too big

File: front/src/applications/operationalStudies/utils.ts [hunk 2]
```diff
@@ -155,35 +158,85 @@ export const formatElectrificationRanges = (
...
+  trainschedulePath: TrainScheduleResult['path'],
```
Comment (clarani) [inline front/src/applications/operationalStudies/utils.ts]: - you forgot a S maj in trainschedule ;) 
- maybe `path` is enough (only a suggestion)
```suggestion
  trainSchedulePath: TrainScheduleResult['path'],
```

File: front/src/applications/operationalStudies/__tests__/utils.spec.ts [hunk 3]
```diff
@@ -124,3 +135,243 @@ describe('isScheduledPointsNotHonored', () => {
...
+            name: 'Mid_East_station',
+            uic: 4,
+          },
+        },
+        position: 26500000,
+      },
+    ]);
+  });
+
+  it('should add waypoints properly even when the last two comes from map click', () => {
```
Comment (clarani) [inline front/src/applications/operationalStudies/__tests__/utils.spec.ts]:
```suggestion
  it('should add waypoints properly even when the last two come from map clicks', () => {
```

File: front/src/applications/operationalStudies/__tests__/sampleData.ts [hunk 4]
```diff
@@ -551,3 +555,136 @@ export const trainSummaryHonored: Extract<SimulationSummaryResult, { status: 'su
...
+    id: '2',
+    offset: 679000,
+    track: 'TC0',
+  },
+  {
+    id: '3',
+    offset: 883000,
+    track: 'TC0',
+  },
+];
```
Comment (clarani) [inline front/src/applications/operationalStudies/__tests__/sampleData.ts]: There is no OP in this path :)

File: front/src/applications/operationalStudies/__tests__/sampleData.ts [hunk 5]
```diff
@@ -551,3 +555,136 @@ export const trainSummaryHonored: Extract<SimulationSummaryResult, { status: 'su
...
+    track: 'TA6',
+  },
+  {
+    id: '2',
+    offset: 4733000,
+    track: 'TA6',
+  },
+];
+
+export const pathInputsContaingOneOperationPoint: TrainScheduleResult['path'] = [
```
Comment (clarani) [inline front/src/applications/operationalStudies/__tests__/sampleData.ts]:
```suggestion
export const pathInputsContaingOneOperationalPoint: TrainScheduleResult['path'] = [
```

File: front/src/applications/operationalStudies/utils.ts (hunk 1 above)
Comment (SharglutDev) [reply front/src/applications/operationalStudies/utils.ts]: Done. I also created a specific file for the tests.

File: front/src/applications/operationalStudies/utils.ts (hunk 2 above)
Comment (SharglutDev) [reply front/src/applications/operationalStudies/utils.ts]: Fine by me for `path` :) Done.

File: front/src/applications/operationalStudies/__tests__/utils.spec.ts (hunk 3 above)
Comment (SharglutDev) [reply front/src/applications/operationalStudies/__tests__/utils.spec.ts]: Done

File: front/src/applications/operationalStudies/__tests__/sampleData.ts (hunk 4 above)
Comment (SharglutDev) [reply front/src/applications/operationalStudies/__tests__/sampleData.ts]: Oh yes it's not very clear, I meant the path will pass through an operational point. I renamed it accordingly to what we test with this sample.

File: front/src/applications/operationalStudies/__tests__/sampleData.ts (hunk 5 above)
Comment (SharglutDev) [reply front/src/applications/operationalStudies/__tests__/sampleData.ts]: Done

File: front/src/applications/operationalStudies/utils.ts (hunk 1 above)
Comment (SharglutDev) [reply front/src/applications/operationalStudies/utils.ts]: I also integrated the changes from the other PR with the `splice`.

**Review by clarani — APPROVED** [review]
LGTM :heavy_check_mark: Thanks for this PR :star_struck:

**Pull request merged**